test/System/String.test.cpp

```diff
@@ -19,7 +19,7 @@ namespace
 template <typename T>
 auto select(const std::basic_string<T>& string16, const std::basic_string<T>& string32)
 {
-    assert(string16 != string32);
+    assert(string16 != string32 && "Invalid to select between identical inputs");
```

Comment:
I don't think this one is a correct sentence. What about something like: "Passing identical arguments to select is invalid."

Reply:
It's just a shorter version of "It is invalid to select between identical inputs". That sounds like correct grammar to me.